plugins/chat/assets/javascripts/discourse/components/chat-thread.js

```diff
@@ -272,6 +274,15 @@ export default class ChatThreadPanel extends Component {
           thread_id: message.thread.staged ? null : message.thread.id,
           staged_thread_id: message.thread.staged ? message.thread.id : null,
         })
+        .then((response) => {
+          if (!this.args.thread.currentUserMembership) {
+            this.args.thread.currentUserMembership =
+              UserChatThreadMembership.create({
+                notification_level: NotificationLevels.TRACKING,
+                last_read_message_id: response.message_id,
+              });
+          }
```

Comment:
This should work I think:

```suggestion
          this.args.thread.currentUserMembership ??=
            UserChatThreadMembership.create({
              notification_level: NotificationLevels.TRACKING,
              last_read_message_id: response.message_id,
            });
```